Plugins/Monero/Services/MoneroConfigurationMigrationService.cs

```diff
@@ -0,0 +1,123 @@
+using System;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+using BTCPayServer.Abstractions.Contracts;
+using BTCPayServer.Logging;
+using BTCPayServer.Plugins.Monero.Configuration;
+
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+
+
+namespace BTCPayServer.Plugins.Monero.Services
+{
+    public class MoneroConfigurationMigrationService : IHostedService
+    {
+        private const string CryptoCode = "XMR";
+        private const string WalletStateMigrationKey = "MoneroWalletStateMigration";
+
+        private readonly ISettingsRepository _settingsRepository;
+        private readonly MoneroRPCProvider _rpcProvider;
+        private readonly Logs _logs;
+
+        public MoneroConfigurationMigrationService(
+            ISettingsRepository settingsRepository,
+            MoneroRPCProvider rpcProvider,
+            Logs logs)
+        {
+            _settingsRepository = settingsRepository;
+            _rpcProvider = rpcProvider;
+            _logs = logs;
+        }
+
+        public async Task StartAsync(CancellationToken cancellationToken)
+        {
+            await MigrateWalletState();
+        }
+
+        public Task StopAsync(CancellationToken cancellationToken)
+        {
+            return Task.CompletedTask;
+        }
+
+        private async Task MigrateWalletState()
```

Comment:
Needs integration test/s.. also other parts of the PR here as you are changing core functionality and increasing complexity